### PR TITLE
Introduce new durability fields & form.

### PR DESCRIPTION
**Pull request by n1k0**
[User Story](https://www.notion.so/D-tailler-dans-la-V2beta-le-calcul-de-l-indice-de-durabilit-1fa1eb1d1a924178bbe56bad2207c8c1)

### CODE REVIEW

**Comment by n1k0** [comment]
I think this is now ready for review, the remaining 3 indicators (still TBD on the card) will be implemented on a next PR.

File: src/Data/Textile/Economics.elm [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    { marketingDuration : Duration
+    , numberOfReferences : Int
+    , price : Price
+    , repairCost : Price
+    }
+
+
+{-| Note: We don't want to specify a specific currency (eg. EUR, USD)
+so we can keep prices abstract yet consistent.
+TODO: move to Unit?
```
Comment (n1k0) [inline src/Data/Textile/Economics.elm:36]: Note: I'm really not sure about that as I really liek the idea of it to be tied to the `Economics` module. Thoughts?

File: src/Data/Textile/Economics.elm [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    = Price Float
+
+
+computeDurabilityIndex :
+    { marketingDuration : Duration
+    , numberOfReferences : Int
+    , price : Price
+    , repairCost : Price
+    }
+    -- FIXME: actually should be a Unit.Ratio?
```
Comment (n1k0) [inline src/Data/Textile/Economics.elm:48]: Note: Same, not sure at all about that, I really like the specificity of a dedicated `Durability` type over a `Ratio`.

File: src/Data/Textile/Economics.elm [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+        finalIndex =
+            [ computeMarketingDurationIndex marketingDuration
+            , computeNumberOfReferencesIndex numberOfReferences
+            , computeRepairCostIndex price repairCost
+            ]
+                |> List.map Unit.ratioToFloat
+                |> List.sum
+                -- FIXME: For now we don't deal with weighting
+                |> (\x -> x / 3)
```
Comment (n1k0) [inline src/Data/Textile/Economics.elm:65]: This will be addressed when we'll have the final 5 or 6 indices, in a next PR.

**Review by n1k0 — COMMENTED** [review]
Adding some context here and there to help reviewing

**Review by magopian — APPROVED** [review]
This all looks great, excellent work!
A few nits and questions, I'll let you decide which of those you want to address.

File: public/data/textile/products.json [hunk 4]
```diff
@@ -6,6 +6,12 @@
     "surfaceMass": 200,
     "yarnSize": 40,
     "fabric": "weaving",
+    "economics": {
+      "marketingDuration": 60,
+      "numberOfReferences": 20000,
+      "price": 20,
+      "repairCost": 10
```
[the remark this answers is not in the document]
Comment (n1k0) [reply public/data/textile/products.json:13]: Nope, it's a fixed value for each product category, not intended to be ever modified by users.

File: src/Page/Textile/Simulator.elm [hunk 5]
```diff
@@ -714,43 +798,34 @@ massField massInput =
...
+            [ text "Indice de durabilité" ]
+        , input
+            [ type_ "range"
+            , id "durability-field"
+            , class "form-range w-auto"
+            , Attr.min (fromFloat Unit.minDurability)
+            , Attr.max (fromFloat Unit.maxDurability)
+
+            -- WARNING: be careful when reordering attributes: for obscure reasons,
+            -- the `value` one MUST be set AFTER the `step` one.
```
[the remark this answers is not in the document]
Comment (n1k0) [reply src/Page/Textile/Simulator.elm:818]: Yeah I agree, we should definitely file a `Tech Debt` card about that.

[the remark this answers is not in the document]
Comment (n1k0) [reply src/Page/Textile/Simulator.elm:818]: A [card](https://www.notion.so/G-rer-l-ordonnancement-automatique-des-attributs-d-inputs-f5391c132fc14fc18334518e372b8ca1) has been created

**Pull request merged**